In [2]:
# Import the packages
# Read the data
# seperate cat and num columns
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

visa_df=pd.read_csv('Visadataset.csv')
cat=visa_df.select_dtypes(include='object').columns
num=visa_df.select_dtypes(exclude='object').columns

In [3]:
cat

Index(['case_id', 'continent', 'education_of_employee', 'has_job_experience',
       'requires_job_training', 'region_of_employment', 'unit_of_wage',
       'full_time_position', 'case_status'],
      dtype='object')

**Continent**

In [4]:
visa_df['continent']

0          Asia
1          Asia
2          Asia
3          Asia
4        Africa
          ...  
25475      Asia
25476      Asia
25477      Asia
25478      Asia
25479      Asia
Name: continent, Length: 25480, dtype: object

**unique**

- Unique will give the lable names which are repeated

In [6]:
visa_df['continent'].unique()

array(['Asia', 'Africa', 'North America', 'Europe', 'South America',
       'Oceania'], dtype=object)

- we have total 25480 values are avialable in continent column 

- but only 6 continents are repeating

**nunique**

In [7]:
visa_df['continent'].nunique()

6

In [8]:
len(visa_df['continent'].unique())

6

**task**

- we know that only 6 lables are repeating entire 25480 only continet column

- now i want to know how many Asia values we have

- now i want to know how many Africa values we have and so on

In [12]:
# step-1: read the column 
visa_df['continent']
# step-2: make equate the the column data = 'Asia'
con=visa_df['continent']=='Asia'
# step-3: We want extract only True data
# main_data_frame[con]
visa_df[con]

case_id continent education_of_employee  ... unit_of_wage full_time_position  case_status
0         EZYV01      Asia           High School  ...         Hour                  Y       Denied
1         EZYV02      Asia              Master's  ...         Year                  Y    Certified
2         EZYV03      Asia            Bachelor's  ...         Year                  Y       Denied
3         EZYV04      Asia            Bachelor's  ...         Year                  Y       Denied
5         EZYV06      Asia              Master's  ...         Year                  Y    Certified
...          ...       ...                   ...  ...          ...                ...          ...
25475  EZYV25476      Asia            Bachelor's  ...         Year                  Y    Certified
25476  EZYV25477      Asia           High School  ...         Year                  Y    Certified
25477  EZYV25478      Asia              Master's  ...         Year                  N    Certified
25478  EZYV25479      Asia              Master's  ...         Year                  Y    Certified
25479  EZYV25480      Asia            Bachelor's  ...         Year                  Y    Certified

[16861 rows x 12 columns]

In [10]:
5==10

False

In [ ]:
visa_df['continent']
con=visa_df['continent']=='Asia'
len(visa_df[con])

In [16]:
visa_df['continent']
con=visa_df['continent']=='Asia'
len(visa_df[con])

visa_df['continent']
con=visa_df['continent']=='Asia'
len(visa_df[con])

visa_df['continent']
con=visa_df['continent']=='Asia'
len(visa_df[con])

visa_df['continent']
con=visa_df['continent']=='Asia'
len(visa_df[con])

visa_df['continent']
con=visa_df['continent']=='Asia'
len(visa_df[con])

16861

In [19]:
unique=visa_df['continent'].unique()
for i in unique:
    visa_df['continent']
    con=visa_df['continent']==i
    count=len(visa_df[con])
    print(f"The number of applicants from {i} is: {count}")

The number of applicants from Asia is: 16861
The number of applicants from Africa is: 551
The number of applicants from North America is: 3292
The number of applicants from Europe is: 3732
The number of applicants from South America is: 852
The number of applicants from Oceania is: 192


In [20]:
for i in visa_df['continent'].unique():
    condition=visa_df['continent']==i
    print(f'the {i} continent has {len(visa_df[condition])} labels')

the Asia continent has 16861 labels
the Africa continent has 551 labels
the North America continent has 3292 labels
the Europe continent has 3732 labels
the South America continent has 852 labels
the Oceania continent has 192 labels


In [21]:
unique=visa_df['continent'].unique()
count=[]
for i in unique:
    visa_df['continent']
    con=visa_df['continent']==i
    count.append(len(visa_df[con]))
count

[16861, 551, 3292, 3732, 852, 192]

In [22]:
unique

array(['Asia', 'Africa', 'North America', 'Europe', 'South America',
       'Oceania'], dtype=object)

**frequency table**

- create a dataframe using count and unique list

- the dataframe has two columns 1.Continent 2.No.Of Applicants

In [24]:
cols=['Continent','No.Of Applicants']
pd.DataFrame(zip(unique,count),
            c
             olumns=cols)

Continent  No.Of Applicants
0           Asia             16861
1         Africa               551
2  North America              3292
3         Europe              3732
4  South America               852
5        Oceania               192

In [25]:
visa_df['continent'].value_counts()

continent
Asia             16861
Europe            3732
North America     3292
South America      852
Africa             551
Oceania            192
Name: count, dtype: int64

In [ ]:
os.makdir
os.paths,join